### PR TITLE
Add AsteroidExplodesWithSettler testcase

### LINKED ISSUE #144: Asteroid explodes with a settler on it
See documentation (beadando_5) page 16.
Don't forget the test fixture (comm diagram) from page 26.
This is 1 test case.

### PR DESCRIPTION
**Pull request by triszt4n**
Closes #144

### CODE REVIEW

File: java/src/hu/holyoil/skeleton/testcases/AsteroidExplodesWithSettler.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    Settler settler;
+
+    @Override
+    public String Name() {
+        return "Asteroid explodes with a Settler on it";
+    }
+
+    @Override
+    protected void load() {
+        Logger.SetEnabled(true);
```
Comment (vercibar) [inline java/src/hu/holyoil/skeleton/testcases/AsteroidExplodesWithSettler.java]: 👀

**Review by vercibar — CHANGES_REQUESTED** [review]
🤔

File: java/src/hu/holyoil/skeleton/testcases/AsteroidExplodesWithSettler.java (hunk 1 above)
Comment (triszt4n) [reply java/src/hu/holyoil/skeleton/testcases/AsteroidExplodesWithSettler.java]: done

File: java/src/hu/holyoil/neighbour/Asteroid.java [hunk 2]
```diff
@@ -211,7 +211,9 @@ public void Explode() {
         Logger.Return();
 
         Logger.Log(this, "Signaling to crewmates that I am exploding");
-        crewmates.forEach(AbstractCrewmate::ReactToAsteroidExplosion);
+        while (crewmates.size() > 0) {
+            crewmates.get(0).ReactToAsteroidExplosion();
+        }
```
Comment (triszt4n) [inline java/src/hu/holyoil/neighbour/Asteroid.java:216]: Still not the best solution (just tested it in a new test case).

Comment (triszt4n) [reply java/src/hu/holyoil/neighbour/Asteroid.java:216]: I'mma keep on working on this

Comment (triszt4n) [reply java/src/hu/holyoil/neighbour/Asteroid.java:216]: Found it, needs shallow copy!

**Review by vercibar — APPROVED** [review]
😉

**Pull request merged**